feat(i18n): add translation support to adoption insights

rohitkrai03 opened the pull request:
## Hey, I just made a Pull Request!

### Added internationalization (i18n) support to Adoption Insights plugin

- ✅ **Multi-language support**: English (default), German, and French
- ✅ **Translation infrastructure**: Built with Backstage's `@backstage/core-plugin-api/alpha` using `createTranslationRef`, translation resources, and semantic key structures with parameter interpolation
- ✅ **Unit test mocking**: Created reusable translation test utilities (`mockTranslations.ts`) that use actual English strings as single source of truth - can be copied to other plugins


The plugin now supports multiple languages with proper fallbacks and maintains all existing functionality. The translation infrastructure and test patterns established here can be reused across other Backstage plugins.

**Languages added:**
- 🇩🇪 German (de)
- 🇫🇷 French (fr)
- 🇺🇸 English (en) - default


https://github.com/user-attachments/assets/f71c21b8-f992-433c-940c-612237827923



#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [x] Added or Updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

karthikjeeyar reviewed (commented):
Verified the changes locally, I noticed a couple of spots in the tooltip that needs to be translated. Everything else looks good.

Tooltips:

<img width="425" height="268" alt="image" src="https://github.com/user-attachments/assets/008c96e6-1b8f-4f0d-b8a2-0f750cfde351" />

<img width="852" height="204" alt="image" src="https://github.com/user-attachments/assets/f8cfdce2-eb46-4e11-9919-e6f2bf2996ea" />

rohitkrai03 commented on workspaces/adoption-insights/packages/app/e2e-tests/insights.test.ts:
```diff
@@ -110,8 +110,8 @@ test('Rest of the panels have no data', async () => {
     'Top catalog entities',
-    'Top TechDocs',
-    'Searches',
+    'Top techdocs',
```
Updated.

rohitkrai03 commented:
> Verified the changes locally, I noticed a couple of spots in the tooltip that needs to be translated. Everything else looks good.

Fixed the tooltip translations.

ciiay reviewed (commented):
Thanks for the PR 👏 

It works mostly fine, just have two small concerns.
1. Should the date/time format be localized as well?
2. Should the exported CSV file name be translated as well? The downloaded csv file is named in English "active-user.csv"
https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/adoption-insights/plugins/adoption-insights/src/api/index.ts#L200

<img width="1218" height="1084" alt="image" src="https://github.com/user-attachments/assets/f094401f-d04e-4c42-a810-1c985ecb1284" />

ShiranHi commented:
Thank you @rohitkrai03 for this, looks great!
I have a quick piece of feedback on the "Export CSV" button, the padding seems too small as it is too close to the card border.
<img width="600" height="484" alt="image" src="https://github.com/user-attachments/assets/4de534ff-f045-4b31-95e1-af735edb555e" />

rm3l commented on workspaces/adoption-insights/plugins/adoption-insights/src/translations/fr.ts:
```diff
@@ -0,0 +1,92 @@
+const adoptionInsightsTranslationFr = createTranslationMessages({
+  ref: adoptionInsightsTranslationRef,
+  messages: {
+    'page.title': "Insights d'Adoption",
+    'header.title': "Insights d'Adoption",
```
"Insights" is not a French word. I guess this is the translation for "Adoption Insights". To check with PMs and/or Docs, but perhaps something like this would make more sense? Or leave the branded name "Adoption Insights" as is?

```suggestion
    'page.title': "Statistiques d'adoption",
    'header.title': "Statistiques d'adoption",
```

rm3l commented on workspaces/adoption-insights/plugins/adoption-insights/src/translations/fr.ts:
```diff
@@ -0,0 +1,92 @@
+    'common.invalidDateFormat': 'Format de date invalide',
+    'permission.title': 'Permissions manquantes',
+    'permission.description':
+      "Pour voir le plugin Insights d'Adoption, contactez votre administrateur pour obtenir les permissions adoption-insights.events.read.",
```
Similar comment as previously. Either keep the plugin name as is, like so:
```suggestion
      "Pour voir le plugin \"Adoption Insights\", contactez votre administrateur pour obtenir les permissions adoption-insights.events.read.",
```

Or translate:
```suggestion
      "Pour voir le plugin de statistiques d'adoption, contactez votre administrateur pour obtenir les permissions adoption-insights.events.read.",
```

rohitkrai03 replied:
Updated.

rm3l commented on workspaces/adoption-insights/plugins/adoption-insights/src/translations/fr.ts:
```diff
@@ -0,0 +1,92 @@
+    'users.licensed': 'Licenciés',
+    'users.ofTotal': 'sur {{total}}',
+    'users.tooltip':
+      "Définissez le nombre d'utilisateurs sous licence dans app-config.yaml",
```
```suggestion
      "Définissez le nombre d'utilisateurs sous licence dans le fichier de configuration app-config.yaml",
```

Might be clearer to state that this is a config file.

rm3l commented on workspaces/adoption-insights/plugins/adoption-insights/src/translations/fr.ts:
```diff
@@ -0,0 +1,92 @@
+    'filter.all': 'Tous',
+    'filter.selectKind': 'Sélectionner le type',
+    'common.noResults': 'Aucun résultat pour cette plage de dates.',
+    'common.readMore': 'Lire plus',
```
```suggestion
    'common.readMore': 'En savoir plus',
```

More idiomatic IMO

rohitkrai03 replied:
Updated.

rm3l commented on workspaces/adoption-insights/plugins/adoption-insights/src/translations/fr.ts:
```diff
@@ -0,0 +1,92 @@
+    'users.title': "Nombre total d'utilisateurs",
+    'users.haveLoggedIn': 'se sont connectés',
+    'users.loggedInUsers': 'Utilisateurs connectés',
+    'users.licensed': 'Licenciés',
```
```suggestion
    'users.licensed': 'Utilisateurs sous license',
```

`Licencié` might be confusing, as it usually means "fired" or "graduate", or can also mean "license-holder", but in the context of sports or other associations :)
So I'd suggest something clearer.

rm3l reviewed (commented):
A few comments on the French 🇫🇷 translation

rohitkrai03 commented:
@ciiay 
> 1. Should the date/time format be localized as well?
Updated the date and time formatting to be locale aware along with timezone aware.
> 2. Should the exported CSV file name be translated as well? The downloaded csv file is named in English "active-user.csv"
>    https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/adoption-insights/plugins/adoption-insights/src/api/index.ts#L200
Updated to use translated filenames.

ciiay reviewed (approved):
/lgtm

Verified the updated date/time strings and the download file name look good in this PR. Great job 👏

HusneShabbir commented:
/lgtm

karthikjeeyar reviewed (approved):
/approve
/lgtm
